src/cmd/ksh93/sh/io.c

```diff
@@ -215,7 +217,7 @@ bool sh_inuse(Shell_t *shp, int fd) { return fd < shp->gd->lim.open_max && shp->
 void sh_ioinit(Shell_t *shp) {
     filemapsize = 8;
     filemap = (struct fdsave *)malloc(filemapsize * sizeof(struct fdsave));
-    sh_iovalidfd(shp, 16);
+    assert(sh_iovalidfd(shp, 16));
```

Comment:
Never use `assert()` if evaluating the condition has side-effects. The assert call can be optimized out depending on build options. Which in this case means the `sh_iovalidfd()` call would not be made. Instead do `if (!sh_iovalidfd(shp, 16)) abort();`.